Doc/library/threading.rst

```diff
@@ -594,14 +594,19 @@ and "recursion level" in addition to the locked/unlocked state used by primitive
 locks.  In the locked state, some thread owns the lock; in the unlocked state,
 no thread owns it.
 
-To lock the lock, a thread calls its :meth:`~RLock.acquire` method; this
-returns once the thread owns the lock.  To unlock the lock, a thread calls
-its :meth:`~Lock.release` method. :meth:`~Lock.acquire`/:meth:`~Lock.release`
-call pairs may be nested; only the final :meth:`~Lock.release` (the
-:meth:`~Lock.release` of the outermost pair) resets the lock to unlocked and
-allows another thread blocked in :meth:`~Lock.acquire` to proceed.
+To lock the lock, a thread calls its :meth:`~RLock.acquire` method.  To unlock
+the lock, a thread calls its :meth:`~Lock.release` method. Reentrant locks
+support the :ref:`context management protocol <with-locks>` so ```with``` can be used.
 
-Reentrant locks also support the :ref:`context management protocol <with-locks>`.
+RLock's :meth:`~RLock.acquire`/:meth:`~RLock.release` call pairs may be nested,
+unlike Lock's :meth:`~Lock.acquire`/:meth:`~Lock.release`. Only the final
+:meth:`~RLock.release` (the :meth:`~Lock.release` of the outermost pair) resets
+the lock to an unlocked state and allows another thread blocked in
+:meth:`~RLock.acquire` to proceed.
+
+:meth:`~RLock.acquire`/:meth:`~RLock.release` must be in pairs: each acquire
+must have a release in the thread that has acquired the lock. Failing to
+release as many times the lock has required will lead to deadlock.
```

Comment:
wording, perhaps:

... call release as many times as the lock has been acquired can lead to deadlock.

Reply:
Yup; I added this in my suggestion